File: Python/boj/Silver/15651.py
import sys
from itertools import product

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N, M = map(int, input().split(" "))
    numbers = [i for i in range(1, N + 1)]
    products = product(numbers, repeat=M)

    for i in products:
        print(" ".join(map(str, i)))

File: JavaScript/boj/Silver/15651.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const [N, M] = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split(' ')
  .map(Number);

const numbers = [];
const result = [];

const dfs = (depth) => {
  if (depth === M) {
    result.push(numbers.join(' '));
    return;
  }

  for (let i = 1; i <= N; i++) {
    numbers.push(i);
    dfs(depth + 1);
    numbers.pop();
  }
};

dfs(0);
console.log(result.join('\n'));
